File: PerfTest2/PerfTest2-Java/app/src/main/java/com/vandammeford/kevinf/perftest2_java/SqLiteTableFragment.java
package com.vandammeford.kevinf.perftest2_java;

import android.app.Activity;
import android.net.Uri;
import android.os.Bundle;
import android.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListView;

public class SqLiteTableFragment extends Fragment {

    private OnFragmentInteractionListener mListener;
    private SqLiteDisplayType displayType;

    public SqLiteTableFragment() {
        // Required empty public constructor
    }

    @Override
    public void onStart() {
        super.onStart();

        ListView lstSqLiteTable = (ListView)this.getActivity().findViewById(R.id.lstSqLiteTable);
        SqLiteTableAdapter adapter = new SqLiteTableAdapter(this.getActivity(), displayType);
        lstSqLiteTable.setAdapter(adapter);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle bundle = getArguments();
        displayType = (SqLiteDisplayType)bundle.getSerializable("displayType");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_sq_lite_table, container, false);
    }

    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (OnFragmentInteractionListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        public void onFragmentInteraction(Uri uri);
    }

}

File: PerfTest2/PerfTest2-Xamarin/PerfTest2-ClassicXamarin.Android/Fragments/SqLiteTableFragment.cs
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using PerfTest2Xamarin.Adapters;
using PerfTest2Xamarin.Enums;

namespace PerfTest2Xamarin.Fragments
{
    public class SqLiteTableFragment : Fragment
    {
        private SqLiteDisplayType displayType;
        private string directory;

        public SqLiteTableFragment() : base()
        {
            // Required empty public constructor
        }

        public SqLiteTableFragment(string directory)
            : base()
        {
            this.directory = directory;
        }

        public override void OnStart()
        {
            base.OnStart();
            ListView lstSqLiteTable = (ListView)Activity.FindViewById(Resource.Id.lstSqLiteTable);
            var adapter = new SqLiteTableAdapter(Activity, directory, displayType);
            lstSqLiteTable.Adapter = adapter;
        }

        public override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            var bundle = this.Arguments;
            displayType = (SqLiteDisplayType)bundle.GetInt("displayType");
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            return inflater.Inflate(Resource.Layout.fragment_sq_lite_table, container, false);
        }

    }
}